src/scripts/sections/product.js

```diff
@@ -6,217 +6,202 @@
-    if (typeof newImageId === 'undefined') {
-      newImageId = $(
-        `${selectors.productImageWrapper}:not('.${cssClasses.hide}')`,
-      ).data('image-id');
+    if (!variant) {
+      submitButton.disabled = true;
+      submitButtonText.innerText = theme.strings.unavailable;
+    } else if (variant.available) {
+      submitButton.disabled = false;
+      submitButtonText.innerText = theme.strings.addToCart;
+    } else {
+      submitButton.disabled = true;
+      submitButtonText.innerText = theme.strings.soldOut;
     }
+  },
 
-    const $thumbnail = $(
-      `${selectors.productThumbs}[data-thumbnail-id='${newImageId}']`,
-    );
-
-    $(selectors.productThumbs)
-      .removeClass(cssClasses.activeThumbnail)
-      .removeAttr('aria-current');
+  renderImages(variant) {
+    if (!variant || variant.featured_image === null) {
+      return;
+    }
 
-    $thumbnail.addClass(cssClasses.activeThumbnail);
-    $thumbnail.attr('aria-current', true);
+    this.renderFeaturedImage(variant.featured_image.id);
+    this.renderActiveThumbnail(variant.featured_image.id);
   },
 
-  switchImage(imageId) {
-    const $newImage = $(
-      `${selectors.productImageWrapper}[data-image-id='${imageId}']`,
-      this.$container,
+  renderPrice(variant) {
+    const priceElement = this.container.querySelector(selectors.productPrice);
+    const priceWrapperElement = this.container.querySelector(
+      selectors.priceWrapper,
     );
-    const $otherImages = $(
-      `${selectors.productImageWrapper}:not([data-image-id='${imageId}'])`,
-      this.$container,
-    );
-    $newImage.removeClass(cssClasses.hide);
-    $otherImages.addClass(cssClasses.hide);
-  },
 
-  /**
-   * Updates the DOM state of the add to cart button
-   *
-   * @param {boolean} enabled - Decides whether cart is enabled or disabled
-   * @param {string} text - Updates the text notification content of the cart
-   */
-  updateAddToCartState(evt) {
-    const variant = evt.variant;
+    priceWrapperElement.classList.toggle(classes.hide, !variant);
 
     if (variant) {
-      $(selectors.priceWrapper, this.$container).removeClass(cssClasses.hide);
-    } else {
-      $(selectors.addToCart, this.$container).prop('disabled', true);
-      $(selectors.addToCartText, this.$container).html(
-        theme.strings.unavailable,
-      );
-      $(selectors.priceWrapper, this.$container).addClass(cssClasses.hide);
+      priceElement.innerText = formatMoney(variant.price, theme.moneyFormat);
+    }
+  },
+
+  renderComparePrice(variant) {
+    if (!variant) {
       return;
     }
 
-    if (variant.available) {
-      $(selectors.addToCart, this.$container).prop('disabled', false);
-      $(selectors.addToCartText, this.$container).html(theme.strings.addToCart);
+    const comparePriceElement = this.container.querySelector(
+      selectors.comparePrice,
+    );
+    const compareTextElement = this.container.querySelector(
+      selectors.comparePriceText,
+    );
```

Comment:
If either of these elements are `null`, meaning if there's no compare price, the code below (e.g. `.innerText`) will throw an error. We'll want to check to see if there's elements exist before proceeding with the updates.

Reply:
@martinamarien I'll need your help to reproduce this. Can't on my end.